### PR TITLE
RHOAIENG-7217 KServe and KServe.Serving management states

### PR DESCRIPTION
**Pull request by mattmahoneyrh**
New test cases:
- DSC KServe Managed and Removed states
- DSC KServe.Serving Removed state

Enhancements:
 - Serverless Operator test cases restore management state to original value
 - Test case 'Validate DSC creates all Serverless CRs' (i.e. DSC Kserve.serving Managed state): Wait for resources to be ready, as to ensure that adequate time is provided for resources to be present

### CODE REVIEW

**Comment by mattmahoneyrh** [comment]
Passed Jenkins RHOAI and ODH.

File: ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot [hunk 1]
```diff
@@ -5,45 +5,137 @@ Library         OpenShiftLibrary
...
+    IF    "${management_state}" != "Removed"
+            Set Component State    kserve/serving    Removed
+    END
+
+    Wait Until Keyword Succeeds    5 min    0 sec
+    ...    Is Resource Present     KnativeServing    knative-serving    ${KNATIVESERVING_NS}   ${IS_NOT_PRESENT}
+
+    Wait For DSC Conditions Reconciled    ${OPERATOR_NAMESPACE}    default-dsc
+
+    Wait Until Keyword Succeeds    2 min    0 sec
```
Comment (CFSNM) [inline ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot:80]: I would set 5 minutes as the timeout for all the keywords.

**Review by CFSNM — CHANGES_REQUESTED** [review]

File: ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot (hunk 1 above)
Comment (mattmahoneyrh) [reply ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot]: Changed timeouts to 5 min.

**Review by CFSNM — APPROVED** [review]
lgtm!

File: ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot [hunk 2]
```diff
@@ -5,45 +5,137 @@ Library         OpenShiftLibrary
...
-    ${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
-    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
+    ...    label_selector=${KNATIVE_SERVING_CONTROLLER_LABEL_SELECTOR}    timeout=300
+
+    #${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    ${KNATIVE_SERVING_CONTROLLER_LABEL_SELECTOR}    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
     #${podname}=    Get Pod Name   ${OPERATOR_NAMESPACE}    ${OPERATOR_LABEL_SELECTOR}
     #Verify Pod Logs Do Not Contain    ${podname}    ${OPERATOR_NAMESPACE}    ${regex_pattern}    rhods-operator
-    Wait For DSC Conditions Reconciled    ${KNATIVESERVING_NS}    default-dsc
+
+Validate DSC Knative Serving Removed State
```
Comment (MarianMacik) [inline ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot:66]: I think this should be
```suggestion
Validate DSC KServe Serving Removed State
```
right?

File: ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot [hunk 3]
```diff
@@ -5,45 +5,137 @@ Library         OpenShiftLibrary
...
+    Wait Until Keyword Succeeds    5 min    0 sec
+    ...    Resource Should Exist     deployment    controller     ${KNATIVESERVING_NS}
+
     Wait For Pods Numbers  2    namespace=${KNATIVESERVING_NS}
-    ...    label_selector=app.kubernetes.io/component=controller    timeout=120
-    ${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
-    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
+    ...    label_selector=${KNATIVE_SERVING_CONTROLLER_LABEL_SELECTOR}    timeout=300
+
+    #${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    ${KNATIVE_SERVING_CONTROLLER_LABEL_SELECTOR}    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
```
Comment (MarianMacik) [inline ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot:62]: As discussed in a chat, this shouldn't be commented.

**Review by MarianMacik — APPROVED** [review]
Looks good now, thanks!

**Review by CFSNM — APPROVED** [review]

**Pull request merged**